esp-hal/src/i2s/master/camera.rs

```diff
@@ -0,0 +1,461 @@
+                // Enable camera mode.
+                .camera_en()
+                .set_bit()
+        });
+
+        // Configure clock divider
+        regs.clkm_conf().modify(|_, w| unsafe {
+            w.clkm_div_a()
+                .bits(0) // Fractional clock divider’s denominator value (6 bits)
+                .clkm_div_b()
+                .bits(0) // Fractional clock divider’s numerator value. (6 bits)
+                .clkm_div_num()
+                .bits(2) // I2S clock divider’s integral value. (8 bits)
+        });
+
+        regs.fifo_conf().modify(|_, w| unsafe {
+            // Enable I2S DMA mode.
+            w.dscr_en()
+                .set_bit()
+                // Receive FIFO mode configuration bit. (3 bits)
+                .rx_fifo_mod()
+                .bits(config.data_format as _)
+                // The bit should always be set to 1.
+                .rx_fifo_mod_force_en()
+                .set_bit()
+        });
+
+        regs.conf_chan().modify(|_, w| unsafe {
+            // I2S receiver channel mode configuration bit. (2 bits)
+            w.rx_chan_mod().bits(1)
+        });
+
+        // Configure the bit length of I2S receiver channel. (6 bits)
+        regs.sample_rate_conf()
+            .modify(|_, w| unsafe { w.rx_bits_mod().bits(16) });
+
+        // Synchronize signals into the receiver in double sync method.
+        regs.timing().write(|w| w.rx_dsync_sw().set_bit());
+
+        // Default these signals to high in case user doesn't provide them.
+        I2S::v_sync_signal().connect_to(Level::High);
+        I2S::h_sync_signal().connect_to(Level::High);
+        I2S::h_enable_signal().connect_to(Level::High);
+
+        let rx_channel = ChannelRx::new(channel.map(|ch| ch.degrade()));
+
+        Self {
+            _i2s: i2s,
+            rx_channel,
+        }
+    }
+}
+
+impl<I2S> Camera<'_, I2S>
+where
+    I2S: RegisterAccess + ExtendedSignals,
+{
+    /// Configures the data pins for the camera interface.
+    /// Use either [`with_data_pins`] *or* [`with_d0`]..[`with_d7`].
+    #[allow(clippy::too_many_arguments)]
+    pub fn with_data_pins<
+        D0: InputPin,
+        D1: InputPin,
+        D2: InputPin,
+        D3: InputPin,
+        D4: InputPin,
+        D5: InputPin,
+        D6: InputPin,
+        D7: InputPin,
+    >(
+        self,
+        d0: impl Peripheral<P = D0>,
+        d1: impl Peripheral<P = D1>,
+        d2: impl Peripheral<P = D2>,
+        d3: impl Peripheral<P = D3>,
+        d4: impl Peripheral<P = D4>,
+        d5: impl Peripheral<P = D5>,
+        d6: impl Peripheral<P = D6>,
+        d7: impl Peripheral<P = D7>,
+    ) -> Self {
+        self.with_d0(d0)
+            .with_d1(d1)
+            .with_d2(d2)
+            .with_d3(d3)
+            .with_d4(d4)
+            .with_d5(d5)
+            .with_d6(d6)
+            .with_d7(d7)
+    }
+
+    /// Configures the d0 pin.
+    /// Use either [`with_data_pins`] *or* [`with_d0`]..[`with_d7`].
```

Comment:
not sure why, but these rustdoc links don't work and i'm not sure how to fix it 😕